crates/ln-dlc-node/src/ln/app_event_handler.rs

```diff
@@ -233,12 +250,66 @@ where
                 payment_hash,
                 onion_fields: _,
                 amount_msat,
-                counterparty_skimmed_fee_msat: _,
+                counterparty_skimmed_fee_msat,
                 purpose,
                 via_channel_id: _,
-                via_user_channel_id: _,
+                via_user_channel_id,
                 claim_deadline: _,
             } => {
+                if counterparty_skimmed_fee_msat > 0 {
+                    tracing::info!("Checking if counterparty skimmed fee msat is justified");
+                    let user_channel_id = via_user_channel_id.context("Missing user channel id")?;
+                    let user_channel_id = UserChannelId::from(user_channel_id);
+                    let channel = self
+                        .node
+                        .storage
+                        .get_channel(&user_channel_id.to_string())?
+                        .with_context(|| {
+                            format!("Couldn't find channel for user_channel_id {user_channel_id}",)
+                        })?;
+
+                    if channel.channel_state == ChannelState::Open {
```

Comment:
❓ 
I don't understand, why would you fail if the channel is already open? 
Wouldn't that prevent you from receiving payments?

Reply:
Because then it is already been paid for.

ChannelReady -> OpenUnpaid
PaymentClaimable -> Open

Reply:
I still don't understand: isn't the event `PaymentClaimable` called for _all_ payments? Meaning, even after the channel was opened paid successfully, if the user tries to receive something, we would fail here?

Reply:
Only if `counterparty_skimmed_fee_msat > 0`, or in other words only if the counterparty is skimming fees from the payment.

Reply:
I guess this scenario means that the coordinator is trying to take extra fees from random payments _after_ the JIT channel has already been created and the opening fee has been paid.

So the app should reject any payment that is taking extra fees arbitrarily. Perhaps eventually it would make sense to show this to the user, so that they can decide whether they want to pay for the special fee or not.